### PR TITLE
pushing role based routing and dropdown in registration

### PR DESCRIPTION
**Pull request by aravindworspace**

### CODE REVIEW

File: codejudge/app/views/registrations/new.html.erb [hunk 1]
```diff
@@ -65,6 +65,16 @@
...
   <% end %>
 
+  <%= form.label :LoginType, "Login Type" %>
+  <%= form.text_field :password_confirmation %>
+  <select name="logintype">
+  <option value=""></option>
+  <option value="2">Instructor</option>
+  <option value="3">Student</option>
+</select>
+<% end %>
```
Comment (shyam-prasad-na) [inline codejudge/app/views/registrations/new.html.erb]: Remove end from here code is breaking with that tag

File: codejudge/app/controllers/welcome_controller.rb [hunk 2]
```diff
@@ -26,7 +26,18 @@ def log_in
...
       session[:user_id] = user.id
-      redirect_to problems_path, notice: 'Logged in successfully'
+      role = Assignment.find_by(user_id: User.find_by(username: params[:username]).id).role_id
+      puts 'printing'
+      puts role
+      if role == 1
+        redirect_to admins_path, notice: 'Logged in admin successfully'
+      end
+      if role == 2
+        redirect_to "http://localhost:3000/problems/1", notice: 'Logged in instructor successfully'
```
Comment (shyam-prasad-na) [inline codejudge/app/controllers/welcome_controller.rb]: Remove hardcoding of value

File: codejudge/app/controllers/welcome_controller.rb [hunk 3]
```diff
@@ -26,7 +26,18 @@ def log_in
...
+      end
+      if role == 4
+          redirect_to instructors_path, notice: 'Logged in student successfully'
```
Comment (shyam-prasad-na) [inline codejudge/app/controllers/welcome_controller.rb]: Note: Always redirecting to instructor page for students too

File: codejudge/app/views/registrations/new.html.erb (hunk 1 above)
Comment (aravindworspace) [reply codejudge/app/views/registrations/new.html.erb]: done

File: codejudge/app/controllers/welcome_controller.rb (hunk 2 above)
Comment (aravindworspace) [reply codejudge/app/controllers/welcome_controller.rb]: removed hard coding

**Review by shyam-prasad-na — APPROVED** [review]
LGTM

**Pull request merged**